### PR TITLE
Added button

### PR DESCRIPTION
**Pull request by vycdev**
Added a switch button;
- nice design
- added it on home page to test it
- removed old button from body that was there to test the usestores

### CODE REVIEW

File: packages/web/src/common/design/components/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from "react"
+
+import {
+    LabelSwitch,
+    InputSwitch,
+    SpanSwitch
+} from "../../../modules/core/styles/switchStyle"
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: Dont use the styles directory

File: packages/web/src/common/design/components/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/*
+    To do:
+    That goddamn interfaace that I dont know how to do it to mention the props (I think)
+*/
+const Switch = props => {
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: type props

File: packages/web/src/common/design/components/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react"
+
+import {
+    LabelSwitch,
+    InputSwitch,
+    SpanSwitch
+} from "../../../modules/core/styles/switchStyle"
+
+/*
+    To do:
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: We use `TODO:`, not `To do:`

File: packages/web/src/common/design/components/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const Switch = props => {
+    return (
+        <LabelSwitch {...props}>
+            <InputSwitch type="checkbox" />
+            <SpanSwitch />
+        </LabelSwitch>
+    )
+}
+
+export default Switch
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: We banned default exporting

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    display: inline-block;
+    width: ${size * 2}px;
+    height: ${size}px;
+`
+export const InputSwitch = styled.input`
+    & {
+        opacity: 0;
+        width: 0;
+        height: 0;
+    }
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: The `& { }` block is useless

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { styled } from "../../theming/themes"
+const size = 30
+const color = "#009fb7"
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: Either use the current theme or add those to the constants.ts

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        position: absolute;
+        cursor: pointer;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        background-color: #ccc;
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: Again, don't use those blocks

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        width: ${(size * 2 - 4) / 2}px;
+        left: 2px;
+        bottom: 1px;
+        background-color: white;
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
+    & {
+        border-radius: 100px;
+    }
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: Once again, theres no need for those blocks

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    & {
+        position: absolute;
+        cursor: pointer;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        background-color: #ccc;
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: The theme has a `durations` prop so you shouldnt hardcode thiese

Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]:
```suggestion
        transition: ${getDuration("short")};
```

File: packages/web/src/common/design/components/Switch.tsx (hunk 2 above)
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: Enforce correct FunctionComponent type
```suggestion
const Switch: React.FC = props => {
```

File: packages/web/src/modules/core/components/Home.tsx [hunk 10]
```diff
@@ -1,14 +1,20 @@
 import React from "react"
 import { useStores } from "../../../common/state/hooks/useStores"
 import { pageCategorires } from "../../metadata/stores/metadataStore"
+import Switch from "../../../common/design/components/Switch"
```
Comment (prescientmoon) [inline packages/web/src/modules/core/components/Home.tsx]: Do not use default imports

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        background-color: ${color};
+    }
+    &:checked + span:before {
+        -webkit-transform: translateX(${(size * 2 - 4) / 2}px);
+        -ms-transform: translateX(${(size * 2 - 4) / 2}px);
+        transform: translateX(${(size * 2 - 4) / 2}px);
+    }
+`
+export const SpanSwitch = styled.span`
+    & {
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: You know you can just remove the & {, right ?

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
+    &:before {
+        position: absolute;
+        content: "";
+        height: ${(size * 2 - 4) / 2}px;
+        width: ${(size * 2 - 4) / 2}px;
+        left: 2px;
+        bottom: 1px;
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: Dont hardcode stuff like that

File: packages/web/src/modules/core/styles/switchStyle.ts [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        background-color: white;
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
+    & {
+        border-radius: 100px;
```
Comment (prescientmoon) [inline packages/web/src/modules/core/styles/switchStyle.ts]: Mord hardcoded stuff

**Review by prescientmoon — CHANGES_REQUESTED** [review]
Maybe next time:)

File: packages/web/src/common/design/components/Switch.tsx (hunk 1 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx]: OK Solved

File: packages/web/src/common/design/components/Switch.tsx (hunk 2 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx]: Solved

File: packages/web/src/common/design/components/Switch.tsx (hunk 3 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx]: Solved

File: packages/web/src/common/design/components/Switch.tsx (hunk 4 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx]: Solved

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 5 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: Almost all of them are gone, some are still required though

File: packages/web/src/common/design/components/Switch.tsx (hunk 2 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx]: done

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 11 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: done

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 12 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: done

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 7 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: done

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 8 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: done

File: packages/web/src/modules/core/styles/switchStyle.ts (hunk 9 above)
Comment (vycdev) [reply packages/web/src/modules/core/styles/switchStyle.ts]: done

File: packages/web/src/modules/core/components/Home.tsx (hunk 10 above)
Comment (vycdev) [reply packages/web/src/modules/core/components/Home.tsx]: done

**Pull request closed without merging**